### PR TITLE
Release 1.0

### PR DESCRIPTION
**Pull request by zlata96**

### CODE REVIEW

File: FunnyPuny/Application/en.lproj/Texts.strings [hunk 1]
```diff
@@ -0,0 +1,62 @@
+//MARK: Home
```
Comment (bystritskiy) [inline FunnyPuny/Application/en.lproj/Texts.strings:1]: Поработать над группировкой разделов. Должны быть разбиты по экранам

File: FunnyPuny/Application/ru.lproj/Texts.strings [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+"wed" = "Ср";
+"thu" = "Чт";
+"fri" = "Пт";
+"sat" = "Сб";
+"sun" = "Вс";
+
+// MARK: Profile
+
+"notifications" = "Уведомления";
+"appearance" = "Appearance";
```
Comment (bystritskiy) [inline FunnyPuny/Application/ru.lproj/Texts.strings]: Appearance?
Мб убрать пока

File: FunnyPuny/Helpers/Extensions/String+Extension.swift [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+        self[min(fromIndex, length) ..< length]
+    }
+
+    func substring(toIndex: Int) -> String {
+        self[0 ..< max(0, toIndex)]
+    }
+
+    subscript(range: Range<Int>) -> String {
+        let fullRange = Range(uncheckedBounds: (lower: max(0, min(length, range.lowerBound)),
+                                                upper: min(length, max(0, range.upperBound))))
```
Comment (bystritskiy) [inline FunnyPuny/Helpers/Extensions/String+Extension.swift]: Поработать над переносами ")"

File: FunnyPuny/Helpers/Extensions/UIStackView+Extension.swift [hunk 4]
```diff
@@ -0,0 +1,31 @@
+// UIStackView+Extension.swift
+// FunnyPuny. Created by Zlata Guseva.
+
+import UIKit
+
+extension UIStackView {
+    func addArrangedSubviews(_ subviews: [UIView]) {
+        for value in subviews {
+            addArrangedSubview(value)
```
Comment (bystritskiy) [inline FunnyPuny/Helpers/Extensions/UIStackView+Extension.swift:9]: переделать через $0

File: .gitignore [hunk 5]
```diff
@@ -2,6 +2,12 @@
 #
```
Comment (bystritskiy) [inline .gitignore:2]: Поработать над .gitignore (ask me)

File: FunnyPuny/BusinessLogic/HabitManager.swift [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+// FunnyPuny. Created by Zlata Guseva.
+
+import Foundation
+import RealmSwift
+import SwiftDate
+
+// swiftlint:disable all
+// TODO:
+class HabitManager {
+    let realm = try! Realm()
```
Comment (bystritskiy) [inline FunnyPuny/BusinessLogic/HabitManager.swift:10]: delete 'swiftlint:disable all'
and put 'swiftlint:disable force_cast'

File: FunnyPuny/Helpers/Extensions/UIStackView+Extension.swift [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+        }
+    }
+
+    func addSeparators(at positions: [Int], color: UIColor) {
```
Comment (bystritskiy) [inline FunnyPuny/Helpers/Extensions/UIStackView+Extension.swift]: not used

File: FunnyPuny/Presentation/AddHabit/AddHabitViewController.swift [hunk 8]
```diff
@@ -6,41 +6,90 @@ import UIKit
...
 class AddHabitViewController: ViewController {
     private var addHabitView = AddHabitView()
+    private let selectedFrequencies = List<Frequency>()
+    private var habitManager = HabitManager()
 
     override func viewDidLoad() {
         super.viewDidLoad()
         title = Texts.addHabit
         view = addHabitView
+        selectedFrequencies.append(objectsIn: [.mon, .tue, .wed, .thu, .fri, .sat, .sun])
```
Comment (bystritskiy) [inline FunnyPuny/Presentation/AddHabit/AddHabitViewController.swift]: why not in 9 line in init?

File: FunnyPuny/Presentation/AddHabit/AddHabitViewController.swift [hunk 9]
```diff
@@ -6,41 +6,90 @@ import UIKit
...
+    }
+}
+
+// MARK: FrequencyViewDelegate
+
+extension AddHabitViewController: FrequencyViewDelegate {
+    func didSelect(_ frequencies: [Frequency]) {
+        selectedFrequencies.removeAll()
+        if frequencies.contains(.everyday) {
+            selectedFrequencies.append(objectsIn: [.mon, .tue, .wed, .thu, .fri, .sat, .sun])
```
Comment (bystritskiy) [inline FunnyPuny/Presentation/AddHabit/AddHabitViewController.swift]: Frequency.allWeek

File: FunnyPuny/Presentation/Analytics/AnalyticsView.swift [hunk 10]
```diff
@@ -0,0 +1,71 @@
+// AnalyticsView.swift
+// FunnyPuny. Created by Zlata Guseva.
+
+import SnapKit
+import UIKit
+
+class AnalyticsView: UIView {
+    var menuButtonView = MenuButtonView()
```
Comment (bystritskiy) [inline FunnyPuny/Presentation/Analytics/AnalyticsView.swift:8]: is it necessary need to be public?

File: FunnyPuny/Presentation/Analytics/AnalyticsViewController.swift [hunk 11]
```diff
@@ -0,0 +1,177 @@
...
+// FunnyPuny. Created by Zlata Guseva.
+
+import JTAppleCalendar
+import RealmSwift
+import SwiftDate
+import UIKit
+
+// MARK: - AnalyticsViewController
+
+enum AnalyticMode {
```
Comment (bystritskiy) [inline FunnyPuny/Presentation/Analytics/AnalyticsViewController.swift]: enum need to be in the class

Comment (bystritskiy) [reply FunnyPuny/Presentation/Analytics/AnalyticsViewController.swift]: and private

File: FunnyPuny/Presentation/Analytics/Calendar/CalendarAnalyticHeaderView.swift [hunk 12]
```diff
@@ -0,0 +1,51 @@
+// CalendarAnalyticHeaderView.swift
+// FunnyPuny. Created by Zlata Guseva.
+
+import JTAppleCalendar
+import SnapKit
+import UIKit
+
+class CalendarAnalyticHeaderView: JTACMonthReusableView {
+    var monthView = CalendarAnalyticHeaderMonthView()
```
Comment (bystritskiy) [inline FunnyPuny/Presentation/Analytics/Calendar/CalendarAnalyticHeaderView.swift:9]: is it necessary need to be public?

File: FunnyPuny/Presentation/Calendar/Analytic/CalendarAnalyticDateCell.swift [hunk 13]
```diff
@@ -0,0 +1,47 @@
...
+        makeConstraints()
+    }
+
+    private func addSubviews() {
+        addSubview(dateLabel)
+    }
+
+    private func makeConstraints() {
+        dateLabel.snp.makeConstraints { make in
+//            make.edges.equalToSuperview()
```
Comment (bystritskiy) [inline FunnyPuny/Presentation/Calendar/Analytic/CalendarAnalyticDateCell.swift:43]: comment

File: FunnyPuny/Presentation/Common /UIElements/HomeViewState.swift [hunk 14]
```diff
@@ -0,0 +1,78 @@
+// HomeViewState.swift
+// FunnyPuny. Created by Zlata Guseva.
+
+import UIKit
+
+class HomeViewState: UIView {
```
Comment (bystritskiy) [inline FunnyPuny/Presentation/Common /UIElements/HomeViewState.swift]: class naming

File: FunnyPuny/Presentation/Common /UIElements/ProgressView.swift [hunk 15]
```diff
@@ -0,0 +1,92 @@
+// ProgressView.swift
+// FunnyPuny. Created by Zlata Guseva.
+
+import UIKit
+
+class ProgressView: UIView {
+    var statusLabel: UILabel = {
```
Comment (bystritskiy) [inline FunnyPuny/Presentation/Common /UIElements/ProgressView.swift:7]: all properties need to be private + to do inner configure

File: FunnyPuny/Presentation/Common /UIElements/ScoreView.swift [hunk 16]
```diff
@@ -0,0 +1,66 @@
+// ScoreView.swift
```
Comment (bystritskiy) [inline FunnyPuny/Presentation/Common /UIElements/ScoreView.swift:1]: same situation like in ProgressView

File: FunnyPuny/Presentation/Home/HomeView.swift [hunk 17]
```diff
@@ -6,20 +6,32 @@ import SnapKit
 import UIKit
 
 class HomeView: UIView {
+    var calendarView: CalendarHomeView = {
+        let view = CalendarHomeView()
+        return view
+    }()
+
+    var emptyImageView = HomeViewState(
```
Comment (bystritskiy) [inline FunnyPuny/Presentation/Home/HomeView.swift]: naming..

File: FunnyPuny/Presentation/Home/HomeViewController.swift [hunk 18]
```diff
@@ -6,14 +6,22 @@ import RealmSwift
 import SwiftDate
 import UIKit
 
-// swiftlint:disable all
+// MARK: - HomeViewController
+
+enum HomeState {
```
Comment (bystritskiy) [inline FunnyPuny/Presentation/Home/HomeViewController.swift]: enum need to be inner in the class + private

File: FunnyPuny/Presentation/Analytics/AnalyticsView.swift (hunk 10 above)
Comment (zlata96) [reply FunnyPuny/Presentation/Analytics/AnalyticsView.swift:8]: yes

File: FunnyPuny/BusinessLogic/HabitManager.swift (hunk 6 above)
Comment (zlata96) [reply FunnyPuny/BusinessLogic/HabitManager.swift:11]: doesn't work

File: FunnyPuny/Presentation/Home/HomeViewController.swift (hunk 18 above)
Comment (zlata96) [reply FunnyPuny/Presentation/Home/HomeViewController.swift]: enum needs to be public, because it is used in home view model

**Review by bystritskiy — APPROVED** [review]

**Pull request merged**